libs/komencikit/src/verifyWallet.ts

```diff
@@ -26,9 +28,21 @@ export const verifyWallet = async (
   expectedSigner: Address
 ): Promise<Result<true, WalletValidationError>> => {
   const code = await contractKit.connection.web3.eth.getCode(metaTxWalletAddress)
+  const eip1167Match = EIP1167_BYTECODE_REGEXP.exec(code)
 
-  if (contractKit.connection.web3.utils.soliditySha3(stripBzz(code)) !== PROXY_BYTECODE_SHA3) {
-    return Err(new InvalidBytecode(metaTxWalletAddress))
+  if (eip1167Match !== null) {
+    // We should have an instance of a EIP1167 Light proxy with a ProxyV2 implementation
+    // So we check the implementation code
+    const proxyImplementationAddress = ensureLeading0x(eip1167Match[1])
+    const proxyByteCode = await contractKit.connection.web3.eth.getCode(proxyImplementationAddress)
+    if (contractKit.connection.web3.utils.soliditySha3(stripBzz(proxyByteCode)) !== PROXY_V2_BYTECODE_SHA3) {
+      return Err(new InvalidBytecode(metaTxWalletAddress))
+    }
+  } else {
+    // We should have an instance of Proxy so we check the bytecode directly
+    if (contractKit.connection.web3.utils.soliditySha3(stripBzz(code)) !== PROXY_V1_BYTECODE_SHA3) {
+      return Err(new InvalidBytecode(metaTxWalletAddress))
+    }
```

Comment:
Do we check the proxy and MTW owners below? I only see checking of the MTW signer